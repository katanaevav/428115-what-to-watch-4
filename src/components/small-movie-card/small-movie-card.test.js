import React from "react";
import renderer from "react-test-renderer";
import SmallMovieCard from "./small-movie-card.jsx";

const Movie = {
  id: 0,
  title: `Aviator`,
  smallPoster: `img/aviator.jpg`,
  preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
};

it(`Render small movie card`, () => {
  const tree = renderer.create(
      <SmallMovieCard
        key = {Movie.id}
        movieId={Movie.id}
        movieTitle = {Movie.title}
        movieSmallPoster = {Movie.smallPoster}
        preview = {Movie.preview}
        onMovieMouseOver = {() => {}}
        onMovieMouseOut = {() => {}}
        onMovieTitleClick = {() => {}}
        renderPlayer = {() => {}}
      />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
