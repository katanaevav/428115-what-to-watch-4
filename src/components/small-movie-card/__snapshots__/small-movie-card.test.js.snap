// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render small movie card 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  data-key={0}
>
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="movie-page.html"
      onClick={[Function]}
    >
      Aviator
    </a>
  </h3>
</article>
`;
